kodi_addon_checker/check_addon.py

```diff
@@ -55,6 +56,9 @@ def start(addon_path, branch_name, all_repo_addons, pr, config=None):
             check_entrypoint.check_complex_addon_entrypoint(
                 addon_report, addon_path, parsed_xml, max_entrypoint_count)
 
+            if config.is_enabled("check_py3_compatibility"):
```

Comment:
We should make all these settings depend on the version branch. And remove the config as far as we can. Or maybe have a proxy config that is inlined in the tool, which resolves branches to individual config settings